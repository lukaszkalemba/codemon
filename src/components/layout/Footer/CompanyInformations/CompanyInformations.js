import React from 'react';

const CompanyInformations = () => (
  <div>
    <a href="tel:+48797709856">+48 797 709 856</a>
    <a href="mailto:biuro@codemon.pl">biuro@codemon.pl</a>
    <p>REGON: 000000000</p>
    <p>KRS: 0000000000</p>
  </div>
);

export default CompanyInformations;
