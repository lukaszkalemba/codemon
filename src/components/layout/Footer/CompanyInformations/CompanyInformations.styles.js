import styled from 'styled-components';

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  opacity: 0;
`;

const A = styled.a`
  font-size: ${({ theme }) => theme.font.size.xs};
  color: ${({ theme }) => theme.color.white};
  text-decoration: none;
  transition: opacity 400ms ease-in-out;
  padding: 12px 0;
  opacity: 0.75;

  :hover {
    opacity: 1;
  }
`;

const Paragraph = styled.p`
  font-size: ${({ theme }) => theme.font.size.xs};
  color: ${({ theme }) => theme.color.white};
  opacity: 0.5;
  padding: 11px 0;
  margin: 0;
`;

export default { Wrapper, A, Paragraph };
