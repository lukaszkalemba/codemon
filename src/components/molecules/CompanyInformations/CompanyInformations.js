import React from 'react';
import PropTypes from 'prop-types';
import S from './CompanyInformations.styles';

const CompanyInformations = ({ animation }) => (
  <S.Wrapper {...animation}>
    <S.A href="tel:+48797709856">+48 797 709 856</S.A>
    <S.A href="mailto:kontakt@lukaszkalemba.pl">kontakt@lukaszkalemba.pl</S.A>
    <S.Paragraph>REGON: 000000000</S.Paragraph>
    <S.Paragraph>KRS: 0000000000</S.Paragraph>
  </S.Wrapper>
);

CompanyInformations.defaultProps = {
  animation: {},
};

CompanyInformations.propTypes = {
  animation: PropTypes.shape({
    variants: PropTypes.shape({
      initial: PropTypes.shape({}),
      animate: PropTypes.shape({}),
    }),
  }),
};

export default CompanyInformations;
