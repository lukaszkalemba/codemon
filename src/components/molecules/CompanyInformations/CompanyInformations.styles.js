import styled from 'styled-components';
import { motion } from 'framer-motion';

const Wrapper = styled(motion.div)`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
`;

const Link = styled.a`
  font-size: ${({ theme }) => theme.font.size.body.s};
  color: ${({ theme }) => theme.color.white};
  text-decoration: none;
  transition: opacity 400ms ease-in-out;
  padding: 13px 0;
  opacity: 0.8;

  ${({ theme }) => theme.media.laptop`
    padding: 12px 0;
  `}
    :hover {
    opacity: 1;
  }
`;

const Paragraph = styled.p`
  font-size: ${({ theme }) => theme.font.size.body.s};
  color: ${({ theme }) => theme.color.white};
  opacity: 0.8;
  padding: 13px 0;
  margin: 0;

  ${({ theme }) => theme.media.laptop`
    padding: 12px 0;
  `}
`;

export default { Wrapper, Link, Paragraph };
