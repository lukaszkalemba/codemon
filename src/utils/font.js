export default {
  size: {
    xxxs: '1rem',
    xxs: '1.3rem',
    xs: '1.6rem',
    s: '2.2rem',
    m: '3rem',
    l: '4rem',
    xl: '5.4rem',
    xxl: '8.5rem',
    xxxl: '10rem',
  },

  weight: {
    regular: '400',
    medium: '500',
    bold: '700',
  },

  family: {
    montserrat: 'Montserrat',
  },
};
