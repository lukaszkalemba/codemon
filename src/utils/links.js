export default [
  {
    id: 0,
    path: '/projekty',
    name: 'projekty',
  },
  {
    id: 1,
    path: '/blog',
    name: 'blog',
  },
  {
    id: 2,
    path: '/wycena',
    name: 'wycena',
  },
];
