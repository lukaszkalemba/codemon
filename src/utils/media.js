export default {
  phone: '576px',
  tablet: '768px',
  desktop: '1150px',
  largeDesktop: '1600px',
};
