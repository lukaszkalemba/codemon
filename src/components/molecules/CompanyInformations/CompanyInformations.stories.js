import React from 'react';
import CompanyInformations from './CompanyInformations';

const variants = {
  childrenVariants: {
    animate: {
      opacity: 1,
      y: 0,
      transition: { duration: 0.8 },
    },
    initial: { opacity: 0, y: 20 },
  },
};

export const companyInformations = () => (
  <CompanyInformations variants={variants} />
);

export default {
  component: CompanyInformations,
  title: 'Molecules|CompanyInformations',
};
