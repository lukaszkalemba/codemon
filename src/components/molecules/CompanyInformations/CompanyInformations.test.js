import React from 'react';
import { render } from 'utils/test-utils';
import CompanyInformations from './CompanyInformations';

describe('<CompanyInformations />', () => {
  it('matches snapshot', () => {
    const { container } = render(<CompanyInformations />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
