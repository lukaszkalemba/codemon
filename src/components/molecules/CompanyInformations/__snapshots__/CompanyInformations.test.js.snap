// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CompanyInformations /> renders correctly and matches snapshot 1`] = `
<div
  class="sc-bdnylx bVpGCt"
>
  <a
    class="sc-gtssRu dWkJeQ"
    href="tel:+48797709856"
  >
    +48 797 709 856
  </a>
  <a
    class="sc-gtssRu dWkJeQ"
    href="mailto:kontakt@lukaszkalemba.pl"
  >
    kontakt@lukaszkalemba.pl
  </a>
  <p
    class="sc-dlnjPT ffojLO"
  >
    REGON: 000000000
  </p>
  <p
    class="sc-dlnjPT ffojLO"
  >
    KRS: 0000000000
  </p>
</div>
`;
